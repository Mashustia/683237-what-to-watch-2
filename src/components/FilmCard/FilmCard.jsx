import React from 'react';
import PropTypes from 'prop-types';

import {getImgName} from '../../utils/utils';

const FilmCard = (props) => {
  const {film, onClick, onFilmFocus, id} = props;

  return (
    <article className='small-movie-card catalog__movies-card' onMouseEnter={() => onFilmFocus(id)}>
      <div className='small-movie-card__image'>
        <img src={`img/${getImgName(film)}.jpg`} alt={film} width='280' height='175'/>
      </div>
      <h3 className='small-movie-card__title' onClick={onClick}>
        <a className='small-movie-card__link' href='movie-page.html'>{film}</a>
      </h3>
    </article>
  );
};

FilmCard.propTypes = {
  film: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  onFilmFocus: PropTypes.func.isRequired
};

export default FilmCard;
