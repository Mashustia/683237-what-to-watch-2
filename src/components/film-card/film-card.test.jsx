import React from 'react';
import renderer from 'react-test-renderer';

import FilmCard from './film-card';

const film = {
  id: 1,
  name: `name`,
  // eslint-disable-next-line camelcase
  preview_image: `123.jpg`,
  // eslint-disable-next-line camelcase
  preview_video_link: `123link`
};

it(`FilmCard component renders correctly`, () => {
  const component = renderer
    .create(<FilmCard
      film={film}
      onFilmFocus={jest.fn()}
      onClick={jest.fn()}
      isVideoActive={false}
      onFilmLeave={jest.fn()}
    />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
