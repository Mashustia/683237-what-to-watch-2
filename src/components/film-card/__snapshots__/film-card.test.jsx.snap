// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard component renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseOut={[MockFunction]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Robocop"
      height="175"
      src="img/robocop.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[MockFunction]}
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Robocop
    </a>
  </h3>
  <div
    className="film-hide"
  >
    <video
      className="preview"
      controls={true}
      src="test"
    />
  </div>
</article>
`;
